sim/hunter/hunter.go

```diff
@@ -79,53 +90,55 @@ func (hunter *Hunter) GetHunter() *Hunter {
-	// if hunter.Talents.TrueshotAura {
-	// 	raidBuffs.TrueshotAura = true
-	// }
-	// if hunter.Talents.FerociousInspiration == 3 && hunter.pet != nil {
-	// 	raidBuffs.FerociousInspiration = true
-	// }
+	if hunter.Talents.TrueshotAura {
+		raidBuffs.TrueshotAura = true
+	}
+	if hunter.Talents.FerociousInspiration && hunter.Pet != nil {
+		raidBuffs.FerociousInspiration = true
+	}
 }
 func (hunter *Hunter) AddPartyBuffs(_ *proto.PartyBuffs) {
 }
+func (hunter *Hunter) CalculateMasteryPoints() float64 {
+	return hunter.GetStat(stats.Mastery) / core.MasteryRatingPerMasteryPoint
+}
+func (hunter *Hunter) CritMultiplier(isRanged bool, isMFDSpell bool, doubleDipMS bool) float64 {
+	primaryModifier := 1.0
+	secondaryModifier := 0.0
+
+	// if isRanged {
+	// 	//secondaryModifier += mortalShotsFactor * float64(hunter.Talents.MortalShots)
+	// 	if isMFDSpell {
+	// 		//secondaryModifier += 0.02 * float64(hunter.Talents.MarkedForDeath)
+	// 	}
+	// }
+
+	return hunter.MeleeCritMultiplier(primaryModifier, secondaryModifier)
+}
 
 func (hunter *Hunter) Initialize() {
-	// Update auto crit multipliers now that we have the targets.
-	// hunter.AutoAttacks.MHConfig().CritMultiplier = hunter.critMultiplier(false, false, false)
-	// hunter.AutoAttacks.OHConfig().CritMultiplier = hunter.critMultiplier(false, false, false)
-	// hunter.AutoAttacks.RangedConfig().CritMultiplier = hunter.critMultiplier(false, false, false)
-
-	// hunter.registerAspectOfTheDragonhawkSpell()
-	// hunter.registerAspectOfTheViperSpell()
-
-	// multiShotTimer := hunter.NewTimer()
-	// arcaneShotTimer := hunter.NewTimer()
-	// fireTrapTimer := hunter.NewTimer()
-
-	// hunter.registerAimedShotSpell(multiShotTimer)
-	// hunter.registerArcaneShotSpell(arcaneShotTimer)
-	// hunter.registerBlackArrowSpell(fireTrapTimer)
-	// hunter.registerChimeraShotSpell()
-	// hunter.registerExplosiveShotSpell(arcaneShotTimer)
-	// hunter.registerExplosiveTrapSpell(fireTrapTimer)
-	// hunter.registerKillShotSpell()
-	// hunter.registerMultiShotSpell(multiShotTimer)
-	// hunter.registerRaptorStrikeSpell()
-	// hunter.registerScorpidStingSpell()
-	// hunter.registerSerpentStingSpell()
-	// hunter.registerSilencingShotSpell()
-	// hunter.registerSteadyShotSpell()
-	// hunter.registerVolleySpell()
-
-	// hunter.registerKillCommandCD()
-	// hunter.registerRapidFireCD()
-
-	// if hunter.Options.UseHuntersMark {
-	// 	hunter.RegisterPrepullAction(0, func(sim *core.Simulation) {
-	// 		huntersMarkAura := core.HuntersMarkAura(hunter.CurrentTarget, hunter.Talents.ImprovedHuntersMark, hunter.HasMajorGlyph(proto.HunterMajorGlyph_GlyphOfHuntersMark))
-	// 		huntersMarkAura.Activate(sim)
-	// 	})
-	// }
+	hunter.AutoAttacks.MHConfig().CritMultiplier = hunter.CritMultiplier(false, false, false)
+	hunter.AutoAttacks.OHConfig().CritMultiplier = hunter.CritMultiplier(false, false, false)
+	hunter.AutoAttacks.RangedConfig().CritMultiplier = hunter.CritMultiplier(false, false, false)
+	hunter.FireTrapTimer = hunter.NewTimer()
```

Comment:
this fire trap timer does not seem shared so it can be removed from here and just created in the fire trap?

Reply:
Black Arrow uses it